trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/GravitinoMetadata.java

```diff
@@ -325,4 +363,175 @@ public void setColumnComment(
         gravitinoColumnHandle.getColumnName(),
         commentString);
   }
+
+  @Override
+  public Optional<JoinApplicationResult<ConnectorTableHandle>> applyJoin(
+      ConnectorSession session,
+      JoinType joinType,
+      ConnectorTableHandle left,
+      ConnectorTableHandle right,
+      ConnectorExpression joinCondition,
+      Map<String, ColumnHandle> leftAssignments,
+      Map<String, ColumnHandle> rightAssignments,
+      JoinStatistics statistics) {
+    if (!(left instanceof GravitinoTableHandle) && !(right instanceof GravitinoTableHandle)) {
```

Comment:
The condition should be `!(left instanceof GravitinoTableHandle) || !(right instanceof GravitinoTableHandle)`?